huntsman/observatory.py

```diff
@@ -476,6 +481,22 @@ def prepare_cameras(self, sleep=60, max_attempts=5, require_all_cameras=False):
                                   f'waiting another {sleep} seconds before checking again.')
                 time.sleep(sleep)
```

Comment:
A more idiomatic alternative implementation:

```
ready_counts = {cam_name: 0 for cam_name in self.cameras.keys()}

for i in range(max_attempts):
    for cam_name, cam in self.cameras.items():
        if cam.is_ready:
            ready_counts[cam_name] += 1
        else:
            ready_counts[cam_name] = 0

    if all([count >= consecutive for count in ready_counts.values()]):
        # All cameras are ready so stop checking now.
        break

    time.sleep(sleep)

failed_cameras = [cam_name for cam_name, count in ready_count.items() if count < consecutive]
if failed_cameras:
    # Handle failed cameras
else:
    # All cameras ready
```

Simplifies things a bit, removing a redundant check of the loop index, replacing some explicit loops with list comprehensions, and removing some unnecessary temporary variables